### ISSUE TITLE
Make webmvc-functional sample work with @SpringBootApplication

### ISSUE THREAD

**Issue by dsyer** [spring-projects-experimental/spring-graalvm-native#307]
Currently it uses `@ImportAutoConfiguration` to manually import a selected set of configs, but breaks on a full-bore auto configuration (the webflux version of the same sample would work fine).

**Comment by dsyer** [comment]
I found I needed this (plus the custom binder and remove request mapping config at build time):

```json
[
	{
		"name": "org.springframework.http.converter.FormHttpMessageConverter",
		"fields": [
			{
				"name": "partConverters"
			}
		]
	},
	{
		"name": "org.springframework.boot.web.servlet.server.Encoding",
		"allDeclaredConstructors": true,
		"allDeclaredMethods": true
	}
]
```

**Comment by dsyer** [comment]
There's still an issue with the /error endpoint in functional mode even with the above JSON.

```
2020-09-25 07:27:24.334  WARN 1 --- [nio-8080-exec-1] o.s.web.servlet.PageNotFound             : No mapping for GET /error
```

**Comment by dsyer** [comment]
See also: https://github.com/spring-projects/spring-boot/issues/23485

**Comment by dsyer** [comment]
Fixed by #311
